src/jmclient/wallet.py

```diff
@@ -280,32 +280,26 @@ def select_utxos(self, mixdepth, amount, utxo_filter=(), select_fn=None,
                             'value': utxos[s['utxo']][1]}
                 for s in selected}
 
-    def get_balance_by_mixdepth(self, max_mixdepth=float('Inf'),
-                                include_disabled=True, maxheight=None):
-        """ By default this returns a dict of aggregated bitcoin
-        balance per mixdepth: {0: N sats, 1: M sats, ...} for all
-        currently available mixdepths.
-        If max_mixdepth is set it will return balances only up
-        to that mixdepth.
+    def get_balance_at_mixdepth(self, mixdepth,
+                     include_disabled=True, maxheight=None):
+        """ By default this returns aggregated bitcoin balance at mixdepth.
         To get only enabled balance, set include_disabled=False.
         To get balances only with a certain number of confs, use maxheight.
         """
-        balance_dict = collections.defaultdict(int)
-        for mixdepth, utxomap in self._utxo.items():
-            if mixdepth > max_mixdepth:
-                continue
-            if not include_disabled:
-                utxomap = {k: v for k, v in utxomap.items(
-                    ) if not self.is_disabled(*k)}
-                if maxheight is not None:
-                    utxomap = {k: v for k, v in utxomap.items(
-                        ) if v[2] <= maxheight}
-            value = sum(x[1] for x in utxomap.values())
-            balance_dict[mixdepth] = value
-        return balance_dict
-
-    def get_utxos_by_mixdepth(self):
-        return deepcopy(self._utxo)
+        utxomap = self._utxo.get(mixdepth)
+        if not utxomap:
+            return 0
+        if not include_disabled:
+            utxomap = {k: v for k, v in utxomap.items(
+                ) if not self.is_disabled(*k)}
+        if maxheight is not None:
+            utxomap = {k: v for k, v in utxomap.items(
+                ) if v[2] <= maxheight}
+        return sum(x[1] for x in utxomap.values())
+
+    def get_utxos_at_mixdepth(self, mixdepth):
```

Comment:
```suggestion
    def get_utxos_at_mixdepth(self, mixdepth: int) -> List[dict]:
```

Reply:
Your suggestion looks wrong to me. This method returns a dictionary mapping `(txid, vout)` tuple to `(path, value, height)` tuple.

Reply:
Also, does something else need to be imported to use these type hints? I know almost nothing about Python, but Google suggests that maybe we need to import pieces from the `typing` module or possibly `from __future__ import annotations`. All of that apparently goes away in Python 3.9, but as far as I can tell, JoinMarket intends to support earlier versions, down to 3.7 if I recall correctly.

Reply:
> Your suggestion looks wrong to me. This method returns a dictionary mapping `(txid, vout)` tuple to `(path, value, height)` tuple.

Maybe I made a mistake, didn't look too carefully. That's why I like to add typehints for new functions, so that it's clear for devs, what function arguments are and what it returns.

> Also, does something else need to be imported to use these type hints?

`from typing import Optional, List, Tuple` (or whatever you need) is all that's needed, just search in existing code for `from typing import`.